src/nnsight/contexts/backends/RemoteBackend.py

```diff
@@ -110,25 +145,27 @@ def __call__(self, obj: RemoteMixin):
 
             request = None
 
+            # If self.job_id is empty, it means were sending a new job.
             if not self.job_id:
 
-                request = self.request(obj)
+                request = self.request()
+                
+            # Otherwise we are getting the status / result of the existing job.
+            self.non_blocking_request(request)
 
-            self.non_blocking_request(request=request)
+        # Cleanup
+        self.object.remote_backend_cleanup()
 
-        obj.remote_backend_cleanup()
-
-    def handle_response(self, data: Any) -> "ResponseModel":
+    def handle_response(self, response: "ResponseModel") -> None:
```

Comment:
For readability, I think that `blocking_request()` and `non_blocking_request()` should be placed immediately after `__call__()`